src/http_server.cpp

```diff
@@ -497,6 +497,11 @@ int HttpServer::catch_all_handler(h2o_handler_t *_h2o_handler, h2o_req_t *req) {
 
     //LOG(INFO) << "Init res: " << custom_gen->response << ", ref count: " << custom_gen->response.use_count();
 
+    if(rpath->action == "documents:search" && path_without_query == "/multi_search") {
+        // append api key and ip to metadata
+        request->metadata = api_auth_key_sent + " " + client_ip;
```

Comment:
What if `api_auth_key_sent` contains a space (we don't enforce character check on the API key). We can instead use this format:

```
<length_of_api_key>:<api_key><client_ip>
```

We will read string until `:`, then extract the API key length, then extract the API key based on length and then finally the client_ip (rest of the string).